src/core/decrypt/__tests__/__global__/get_license.test.ts

```diff
@@ -392,31 +396,28 @@ function checkGetLicense(
     contentDecryptor.onInitializationData(initDataEvent);
 
     if (!shouldFail) {
-      const timeout = nbRetries === 0 ? 100 :
-                      nbRetries === 1 ? 300 :
-                      nbRetries === 2 ? 800 :
-                      nbRetries === 3 ? 1200 :
-                      nbRetries === 4 ? 3000 :
-                                        10000;
-      setTimeout(() => {
-        try {
-          if (ignoreLicenseRequests) {
-            expect(mockUpdate).toHaveBeenCalledTimes(0);
-          } else {
-            const license = concat(challenge, challenge);
-            expect(mockUpdate).toHaveBeenCalledTimes(1);
-            expect(mockUpdate).toHaveBeenCalledWith(license);
-          }
-          expect(mockGetLicense).toHaveBeenCalledTimes(nbRetries + 1);
-          for (let i = 1; i <= nbRetries + 1; i++) {
-            expect(mockGetLicense).toHaveBeenNthCalledWith(i, challenge, "license-request");
+      const intervalId = setInterval(() => {
+        if (hasReceivedLicense) {
+          clearInterval(intervalId);
+          try {
+            if (ignoreLicenseRequests) {
+              expect(mockUpdate).toHaveBeenCalledTimes(0);
+            } else {
+              const license = concat(challenge, challenge);
+              expect(mockUpdate).toHaveBeenCalledTimes(1);
+              expect(mockUpdate).toHaveBeenCalledWith(license);
+            }
+            expect(mockGetLicense).toHaveBeenCalledTimes(nbRetries + 1);
+            for (let i = 1; i <= nbRetries + 1; i++) {
+              expect(mockGetLicense).toHaveBeenNthCalledWith(i, challenge, "license-request");
+            }
+            contentDecryptor.dispose();
+            res();
+          } catch (e) {
+            rej(e);
           }
-          contentDecryptor.dispose();
-          res();
-        } catch (e) {
-          rej(e);
         }
-      }, timeout);
+      }, 500);
```

Comment:
As discussed, I find the action of also testing the amount of time after which license-requesting code is re-triggered to be useful. Maybe we can add another - more tolerant - timeout logic?